File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/ObjC/Chat/Chat/ViewController.m

#import "ViewController.h"
#import "ChatViewController.h"
#import <Backendless/Backendless-Swift.h>

@interface ViewController() {
    NSString *userName;
    Channel *channel;
}
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event{
    [self.view endEditing:YES];
}

- (void)showErrorAlert:(Fault *)fault {
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:[NSString stringWithFormat:@"Error %li", fault.faultCode] message:fault.message preferredStyle:UIAlertControllerStyleAlert];
    UIAlertAction *dismissButton = [UIAlertAction actionWithTitle:@"Dismiss" style:UIAlertActionStyleCancel handler:nil];
    [alert addAction:dismissButton];
    [self presentViewController:alert animated:YES completion:nil];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([segue.identifier isEqualToString:@"ShowChat"]) {
        ChatViewController *chatVC = [segue destinationViewController];
        chatVC.userName = userName;
        chatVC.channel = channel;
    }
}

- (IBAction)pressedStartChat:(id)sender {
    __weak ViewController *weakSelf = self;
    if ([self.userNameTextField.text length] > 0) {
        userName = self.userNameTextField.text;
        channel = [Backendless.shared.messaging subscribeWithChannelName:@"realtime_example"];
        RTSubscription *subscription __unused = [channel addConnectListenerWithResponseHandler:^{
            [weakSelf performSegueWithIdentifier:@"ShowChat" sender:nil];
        } errorHandler:^(Fault *fault) {
            [weakSelf showErrorAlert:fault];
        }];
    }
    else {
        [self showErrorAlert:[[Fault alloc] initWithMessage:@"Please enter user name" faultCode:0]];
    }
}

@end
         		

File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/Swift/Chat/Chat/ViewController.swift


import UIKit
import Backendless

class ViewController: UIViewController {
    
    @IBOutlet var userNameTextField: UITextField!
    
    var userName: String?
    var channel: Channel?
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        view.endEditing(true)
    }
    
    func showErrorAlert(_ fault: Fault) {
        let alert = UIAlertController(title: String(format: "Error %@", fault.faultCode), message: fault.message, preferredStyle: .alert)
        let dismissButton = UIAlertAction(title: "Dismiss", style: .cancel, handler: nil)
        alert.addAction(dismissButton)
        present(alert, animated: true, completion: nil)
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if (segue.identifier == "ShowChat") {
            let chatVC = segue.destination as! ChatViewController
            chatVC.userName = userName
            chatVC.channel = channel
        }
    }
    
    @IBAction func pressedStartChat(_ sender: Any) {
        if (!(userNameTextField.text?.isEmpty)!) {
            userName = userNameTextField.text
            channel = Backendless.shared.messaging.subscribe(channelName: "realtime_example")
            let _ = channel?.addConnectListener(responseHandler: {
                self.performSegue(withIdentifier: "ShowChat", sender: nil)
            }, errorHandler: { fault in
                self.showErrorAlert(fault)
            })
        }
        else {
            showErrorAlert(Fault(message: "Please enter user name", faultCode: 0))
        }        
    }
}

